From 59fb2164cb6a71ac17c68cacbc743bf03eeb50e0 Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Wed, 15 Aug 2018 14:16:24 +0800
Subject: [PATCH 268/706] drm: hdp: Correct source core clock for eDP/DP
 controller

This patch corrects source core clock for eDP/DP controller.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit f587723e67c3a06aef29afafee388f0fb62083d9)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 35 +++++++++++--------
 1 file changed, 21 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 73439e6b5440..d1f9815c204c 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -130,6 +130,13 @@
 		clock-output-names= "pclk";
 	};
 
+	coreclk: coreclk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <162500000>;
+		clock-output-names= "coreclk";
+	};
+
 	reboot {
 		compatible ="syscon-reboot";
 		regmap = <&dcfg>;
@@ -621,24 +628,24 @@
 		compatible = "fsl,ls1028a-dp";
 		reg = <0x0 0xf200000 0x0 0x100000>;
 		interrupts = <0 221 0x4>;
-		clocks = <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
+		clocks = <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
 			 <&dpclk>,
 			 <&dpclk>,
 			 <&dpclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
-			 <&pclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
+			 <&coreclk>,
 			 <&dpclk>,
-			 <&pclk>,
-			 <&pclk>;
+			 <&coreclk>,
+			 <&coreclk>;
 		clock-names = "dig_pll", "av_pll", "clk_ipg",
 			      "clk_core", "clk_pxl", "clk_pxl_mux",
 			      "clk_pxl_link", "clk_hdp", "clk_phy",
-- 
2.17.1

